src/main.rs

```diff
@@ -191,175 +191,188 @@ struct Setup {
-            "normalisation-knee",
-            "Knee steepness of the dynamic limiter. Default is 1.0.",
-            "KNEE",
-        )
-        .optopt(
-            "",
-            "volume-ctrl",
-            "Volume control type - [linear, log, fixed]. Default is logarithmic",
-            "VOLUME_CTRL"
-        )
-        .optflag(
-            "",
-            "autoplay",
-            "autoplay similar songs when your music ends.",
-        )
-        .optflag(
-            "",
-            "disable-gapless",
-            "disable gapless playback.",
-        )
-	    .optflag(
-            "",
-            "passthrough",
-            "Pass raw stream to output, only works for \"pipe\"."
-        );
+    .optopt("n", "name", "Device name.", "NAME")
+    .optopt("", "device-type", "Displayed device type.", "TYPE")
+    .optopt(
+        "b",
+        "bitrate",
+        "Bitrate (kbps) {96|160|320}. Defaults to 160.",
+        "BITRATE",
+    )
+    .optopt(
+        "",
+        "onevent",
+        "Run PROGRAM when a playback event occurs.",
+        "PROGRAM",
+    )
+    .optflag("", "emit-sink-events", "Run program set by --onevent before sink is opened and after it is closed.")
+    .optflag("v", "verbose", "Enable verbose output.")
+    .optflag("V", "version", "Display librespot version string.")
+    .optopt("u", "username", "Username to sign in with.", "USERNAME")
+    .optopt("p", "password", "Password", "PASSWORD")
+    .optopt("", "proxy", "HTTP proxy to use when connecting.", "URL")
+    .optopt("", "ap-port", "Connect to AP with specified port. If no AP with that port are present fallback AP will be used. Available ports are usually 80, 443 and 4070.", "PORT")
+    .optflag("", "disable-discovery", "Disable discovery mode.")
+    .optopt(
+        "",
+        "backend",
+        "Audio backend to use. Use '?' to list options.",
+        "NAME",
+    )
+    .optopt(
+        "",
+        "device",
+        "Audio device to use. Use '?' to list options if using alsa, portaudio or rodio.",
+        "NAME",
+    )
+    .optopt(
+        "",
+        "format",
+        "Output format {F32|S32|S24|S24_3|S16}. Defaults to S16.",
+        "FORMAT",
+    )
+    .optopt("", "mixer", "Mixer to use {alsa|softvol}.", "MIXER")
+    .optopt(
+        "m",
+        "mixer-name",
+        "Alsa mixer control, e.g. 'PCM' or 'Master'. Defaults to 'PCM'.",
+        "NAME",
+    )
+    .optopt(
+        "",
+        "mixer-card",
+        "Alsa mixer card, e.g 'hw:0' or similar from `aplay -l`. Defaults to 'default'.",
+        "CARD",
+    )
+    .optopt(
+        "",
+        "mixer-index",
+        "Alsa index of the cards mixer. Defaults to 0.",
+        "INDEX",
+    )
+    .optflag(
+        "",
+        "mixer-linear-volume",
+        "Disable Alsa mixer mapped volume scale (cubic). Defaults to false.",
+    )
+    .optopt(
+        "",
+        "initial-volume",
+        "Initial volume (%) once connected {0..100}. Defaults to 50.",
```

Comment:
The new one is slightly more confusing.

Reply:
Changed to `Initial volume once connected {0-100}%. Defaults to 50 for softvol and for Alsa mixer the current volume.` in 2c9ac40. Better that way?

Reply:
> Initial volume

seems fine

> once connected

what does that mean?

> {0-100}%.

why curly braces. And do I have to add the "%" sign in my input.

> Defaults to 50 for softvol and for Alsa mixer the current volume.

Nice chiasmus. But I'd prefer a more regular syntax.

Reply:
No worries I appreciate the vetting. I had to look up chiasmus. 😄 How about:
`Initial volume in % from 0-100. Default for softvol: '50'. For the Alsa mixer: the current volume.`

Reply:
Sounds good

Reply:
Done in 3e2a71e